### PR TITLE
Cahill-Keyes

### PR DESCRIPTION
**Pull request by Fil**
implementation by Mary-Jo Graça
ported to D3 by @espinielli

### CODE REVIEW

**Comment by Fil** [comment]
@espinielli would you like to review this branch?

File: src/cahillKeyes.js [hunk 1]
```diff
@@ -0,0 +1,713 @@
...
+      normals = (cornerNormals[j] = []);
+    for (var i = 0; i < n; ++i) {
+      ck.push([
+        face[i],
+        hexagon[(i * 2 + 2) % (2 * n)],
+        hexagon[(i * 2 + 1) % (2 * n)]
+      ]);
+      parents.push(j);
+      normals.push(
+        cross(
```
Comment (espinielli) [inline src/cahillKeyes.js]: Maybe use directly `cartesianCross` from `d3-geo`?

File: src/cahillKeyes.js [hunk 2]
```diff
@@ -0,0 +1,713 @@
...
+ * Cahill-Keyes projection
+ *
+ * Implemented in Perl by Mary Jo Graça (2011)
+ *
+ * Ported to D3.js by Enrico Spinielli (2013)
+ *
+ */
+import { abs, asin, atan2, cos, degrees, max, min, pi, radians, sin, sqrt, tan } from "./math";
+import polyhedral from "./polyhedral/index";
+import { geoProjectionMutator as projectionMutator } from "d3-geo";
```
Comment (espinielli) [inline src/cahillKeyes.js]: Re-use code from `d3-geo`
`import { cartesian, spherical } from "d3-geo";`
and remove the L142-L155

File: src/cahillKeyes.js [hunk 3]
```diff
@@ -0,0 +1,713 @@
...
+          }
+        }
+      }
+    }
+    return xy;
+  }
+
+  // from half-octant to megamap (single rotated octant)
+
+  function mj2g(xy, octant) {
```
Comment (espinielli) [inline src/cahillKeyes.js]: This is taken directly from MaryJo implementation but to have it in the spirit of polyhedral map projection, all work of translating/rotating would really need to go in `faceProjection` L17...but I personally have never really had the time and stamina to go thru it...

Comment (Fil) [reply src/cahillKeyes.js]: Yes and I suppose this is what we'll need to do if we want to manipulate the tree of faces for the Antarctica challenge.

File: src/cahillKeyes.js (hunk 1 above)
Comment (Fil) [reply src/cahillKeyes.js]: 👍

File: src/cahillKeyes.js (hunk 2 above)
Comment (Fil) [reply src/cahillKeyes.js]: 👍

**Pull request merged**